components/patina_smbios/src/manager/protocol.rs

```diff
@@ -176,6 +186,39 @@ impl SmbiosProtocol {
             match manager.add_from_bytes(producer_opt, full_record_bytes) {
                 Ok(handle) => {
                     *smbios_handle = handle;
+
+                    // Republish the table with the new record using the two-step deadlock-free approach:
+                    // 1. Build table data into pre-allocated buffer (WITH manager lock)
+                    // 2. Install configuration table (WITHOUT manager lock to avoid deadlock)
+
+                    // Drop manager borrow before republishing
+                    drop(manager);
+
+                    // Rebuild table data into pre-allocated buffer
+                    // The configuration table was already installed during init and points to our
+                    // pre-allocated entry point buffer. We just need to update the data in place.
+                    let (table_addr, ep_addr, entry_point) = {
+                        let manager = match internal.manager.try_borrow() {
+                            Ok(m) => m,
+                            Err(_) => return efi::Status::DEVICE_ERROR,
+                        };
+                        match manager.build_table_data(internal.boot_services) {
+                            Ok((t_addr, e_addr, ep_struct)) => (t_addr, e_addr, ep_struct),
+                            Err(_) => {
+                                return efi::Status::DEVICE_ERROR;
+                            }
+                        }
+                    }; // manager borrow dropped here
+
+                    // Store addresses in manager
+                    {
+                        let manager = match internal.manager.try_borrow() {
```

Comment:
Same comment: try_borrow() should always succeed unless you are designing for a specifically re-entrant scenario (in which case you should describe that scenario in comments for posterity).